### ISSUE TITLE
Setting MaximumNumberOfIterations leads to Exception

### ISSUE THREAD

**Issue by twalthr** [stratosphere/stratosphere#422]
I don't know the maximum number of iterations when creating the plan, so I tried to set the maximum to `Integer.MAX_VALUE` but this leads to 

```
Exception in thread "main" java.lang.IllegalArgumentException: Cannot subtract more cost then there is.
at eu.stratosphere.compiler.costs.Costs.subtractCosts(Costs.java:365)
at eu.stratosphere.compiler.plan.DualInputPlanNode.setCosts(DualInputPlanNode.java:206)
at eu.stratosphere.compiler.costs.CostEstimator.costOperator(CostEstimator.java:204)
at eu.stratosphere.compiler.dag.TwoInputNode.getAlternativePlans(TwoInputNode.java:469)
at eu.stratosphere.compiler.dag.WorksetIterationNode.instantiate(WorksetIterationNode.java:299)
at eu.stratosphere.compiler.dag.TwoInputNode.addLocalCandidates(TwoInputNode.java:514)
at eu.stratosphere.compiler.dag.TwoInputNode.getAlternativePlans(TwoInputNode.java:444)
at eu.stratosphere.compiler.dag.SingleInputNode.getAlternativePlans(SingleInputNode.java:244)
at eu.stratosphere.compiler.dag.DataSinkNode.getAlternativePlans(DataSinkNode.java:220)
at eu.stratosphere.compiler.PactCompiler.compile(PactCompiler.java:708)
at eu.stratosphere.compiler.PactCompiler.compile(PactCompiler.java:543)
at eu.stratosphere.client.LocalExecutor.executePlan(LocalExecutor.java:94)
at eu.stratosphere.client.LocalExecutor.execute(LocalExecutor.java:152)
at eu.stratosphere.test.testPrograms.fullTest.FullTest.main(FullTest.java:71)
```

I'm using:

```
DeltaIteration iteration = new DeltaIteration(0);
iteration.setMaximumNumberOfIterations(Integer.MAX_VALUE);
```

Other high numbers such as `Integer.MAX_VALUE / 4` lead to the same result.

Is this a bug?

**Comment by fhueske** [comment]
Yes, I think that's a bug.
The optimizer weights the cost of operators within iterations by the number of iterations. So, with a very high number of max iteration, that cost becomes huge and will cause the exception.

**Comment by StephanEwen** [comment]
It seems that the NUM_ITERATIONS \* PER_ITERATION_COST leads to a number
overflow. I'll have a look at that.

**Comment by rmetzger** [comment]
fixed in 5e40545ca23c06d9d1e8700e50ac6b881b4fb549
